### PR TITLE
GCS Specific Metrics

### PR DESCRIPTION
**Pull request by guljain**
Added GCS Connector Specific Metrics :
1. gcs_client_side_error_count : Counts the occurrence of client side error status code
2. gcs_server_side_error_count: Counts the occurrence of server side error status code
3. gcs_client_rate_limit_count: Counts the occurrence of 429 status code
4. gcs_total_request_count: Counts the total number of gcs requests made
5. exception_count: Counts the number of exceptions encountered (Can be expanded in future)

### CODE REVIEW

**Comment by guljain** [comment]
Needs /gcbrun

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java [hunk 1]
```diff
@@ -92,6 +109,10 @@ private long increment(GhfsStatistic statistic) {
     return incrementCounter(statistic, 1);
   }
 
+  private long increment(GoogleCloudStorageStatistics statistic) {
```
Comment (arunkumarchacko) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java:112]: Any reason why it is returning value?

File: util/src/main/java/com/google/cloud/hadoop/util/GoogleCloudStorageEventBus.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   *
+   * @param request based on which metrics are updated in Subscriber method
+   */
+  public static void postOnHttpRequest(HttpRequest request) {
+    eventBus.post(request);
+  }
+
+  /** Posting Exception to invoke corresponding Subscriber method. */
+  public static void postOnException() {
+    eventBus.post(exception);
```
Comment (arunkumarchacko) [inline util/src/main/java/com/google/cloud/hadoop/util/GoogleCloudStorageEventBus.java:71]: Why posting a dummy exception? Why not just increment?

Comment (guljain) [reply util/src/main/java/com/google/cloud/hadoop/util/GoogleCloudStorageEventBus.java:74]: updated reason as a comment

**Review by arunkumarchacko — APPROVED** [review]

**Comment by guljain** [comment]
gcbrun

**Pull request merged**